lib/ratchet/component.js

```diff
@@ -52,23 +52,23 @@ var NodeType = {
   TEXT: 3,
 };
 
+function applyArray(node, data) {
+  return data.map((d) => apply(node, d));
+}
+
 function applyText(node, data) {
   var content = [data];
   return {content};
 }
 
-function applyArray(node, data) {
-  return data.map((d) => apply(node, d));
-}
-
 function applyObject(node, data) {
   var content = Array.prototype.map.call(node.childNodes, (child) => createComp(child, data));
   return {content};
 }
 
 function apply(node, data) {
-  if (typeof data === 'string') { return applyText(node, data) }
   if (Array.isArray(data)) { return applyArray(node, data) }
+  if (typeof data === 'string') { return applyText(node, data) }
```

Comment:
A lot of JS libs provide wrappers for type checking to get things like this consistent. I like that you aren't using one of those libs since it's an unnecessary dependency, but there could be readability value in some `is_string`, `is_array` helpers to make sure your code is consistent. Wouldn't want to end up using a different type of type check somewhere else, esp. since type checking in JS is so cray.